woocommerce/woocommerce-gutenberg-products-block
# BlockTemplateController: adds the ability to load and manage block template files

Added `BlockTemplateController` which will give us the ability to load default block templates in the future. In this specific PR the class is checking if they exist in the theme before loading them from Woo Blocks `templates/block-templates/` directory.


Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4969

### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Ensure you're using a block enabled theme
2. Add the **second** template below to the Woo Blocks path specified above the snippet.
3. Confirm this loads in the Site Editor templates, it should say in the editor "You are using the Woo Blocks block template now"
4. Add the **first** template below to themes directory path specified above the snippet.
5. Confirm this loads in the Site Editor templates, it should say in the editor "You are using the Themes block template now"
6. This code is not yet responsible for loading the templates on the frontend. This will be done as part of https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4969

`/theme-dir/block-templates/single-product.html`
```html

 <p>You are using the Themes block template now</p> 

```

`/woo-blocks/templates/block-templates/single-product.html`
```html

 <p>You are using the Woo Blocks block template now</p> 

```

### Changelog

> BlockTemplateController: adds the ability to load and manage block template files.
